sqlcapture/capture.go

```diff
@@ -191,224 +203,80 @@ func (c *Capture) Run(ctx context.Context) (err error) {
-			primaryKey = binding.Resource.PrimaryKey
-			logrus.WithFields(logrus.Fields{"stream": streamID, "key": primaryKey}).Debug("using resource primary key")
-		} else if len(binding.CollectionKey) > 0 {
-			for _, ptr := range binding.CollectionKey {
-				primaryKey = append(primaryKey, collectionKeyToPrimaryKey(ptr))
-			}
-			logrus.WithFields(logrus.Fields{"stream": streamID, "key": primaryKey}).Debug("using collection primary key")
-		} else if len(discoveryInfo.PrimaryKey) > 0 {
-			primaryKey = discoveryInfo.PrimaryKey
-			logrus.WithFields(logrus.Fields{"stream": streamID, "key": primaryKey}).Warn("using discovery primary key -- this is DEPRECATED and also shouldn't be possible")
-		} else {
-			return fmt.Errorf("stream %q: primary key must be specified", streamID)
-		}
 
 		// See if the stream is already initialized. If it's not, then create it.
 		var streamState, ok = c.State.Streams[stateKey]
-		if !ok || streamState.Mode == TableModeIgnore {
-			c.State.Streams[stateKey] = &TableState{Mode: TableModePending, KeyColumns: primaryKey, dirty: true}
-			continue
-		}
-		allStreamsAreNew = false
-
-		// The following safety checks only matter if we're actively backfilling the table or
-		// intend to start backfilling it shortly. But checking for that would risk silently
-		// failing if a new keyed backfill mode were added in the future, so instead we check
-		// for the set of known-to-be-fine situations here.
-		var notBackfilling = (streamState.Mode == TableModeIgnore) ||
-			(streamState.Mode == TableModeActive) ||
-			(streamState.Mode == TableModeKeylessBackfill) ||
-			(streamState.Mode == TableModePending && binding.Resource.Mode == BackfillModeOnlyChanges) ||
-			(streamState.Mode == TableModePending && binding.Resource.Mode == BackfillModeWithoutKey)
-		if notBackfilling {
+		if ok && streamState.Mode != TableStateIgnore {
+			allStreamsAreNew = false
 			continue
 		}
 
-		// Print a warning if the primary key we'll be using differs from the database's primary key.
-		if strings.Join(primaryKey, ",") != strings.Join(discoveryInfo.PrimaryKey, ",") {
-			logrus.WithFields(logrus.Fields{
-				"stream":      streamID,
-				"backfillKey": primaryKey,
-				"databaseKey": discoveryInfo.PrimaryKey,
-			}).Warn("primary key for backfill differs from database table primary key")
-		}
-
-		// Error out if the stream state was previously initialized with a different key
-		if strings.Join(streamState.KeyColumns, ",") != strings.Join(primaryKey, ",") {
-			return fmt.Errorf("stream %q: primary key %q doesn't match initialized scan key %q", streamID, primaryKey, streamState.KeyColumns)
-		}
+		logrus.WithField("stateKey", stateKey).Info("binding added to capture")
+		c.State.Streams[stateKey] = &TableState{Mode: TableStatePending, dirty: true}
 	}
 
-	// Likewise streams may be removed from the catalog, and we need to forget
-	// the corresponding state information.
+	// When a binding is removed we change the table state to "Ignore
```

Comment:
Missing characters at the end of this comment?

Reply:
Good catch, fixed.